core/src/test/scala/zio/config/CoproductTest.scala

```diff
@@ -119,7 +120,7 @@ object CoproductTestUtils {
     case class I(i: Double)           extends Z
   }
 
-  final case class Ldap(value: String) extends AnyVal
+  final case class Ldap(value: String)
```

Comment:
I can't remember, fo we have a special consideration for AnyVal in magnolia. I don't remember myself doing it though. Anyway cool!

Reply:
I had to remove `AnyVal` because the Scala3 product mirror does not work with it. People not interested in Scala3 migration can still use it, with the Scala2 macro it works.